migration-tool/src/main/java/software/amazon/awssdk/migration/internal/recipe/S3StreamingRequestToV2.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.migration.internal.recipe;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.Tree;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JContainer;
+import org.openrewrite.java.tree.JRightPadded;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.Space;
+import org.openrewrite.java.tree.TypeUtils;
+import org.openrewrite.marker.Markers;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.migration.internal.utils.IdentifierUtils;
+
+@SdkInternalApi
+public class S3StreamingRequestToV2 extends Recipe {
+    private static final MethodMatcher PUT_OBJECT =
+        new MethodMatcher("com.amazonaws.services.s3.AmazonS3 "
+                          + "putObject(java.lang.String, java.lang.String, java.io.File)",
+                          true);
+    private static final JavaType.FullyQualified V1_PUT_OBJECT_REQUEST =
+        TypeUtils.asFullyQualified(JavaType.buildType("com.amazonaws.services.s3.model.PutObjectRequest"));
+    private static final JavaType.FullyQualified REQUEST_BODY =
+        TypeUtils.asFullyQualified(JavaType.buildType("software.amazon.awssdk.core.sync.RequestBody"));
+
+    @Override
+    public String getDisplayName() {
+        return "S3StreamingRequestToV2";
+    }
+
+    @Override
+    public String getDescription() {
+        return "S3StreamingRequestToV2.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new Visitor();
+    }
+
+    private static final class Visitor extends JavaIsoVisitor<ExecutionContext> {
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext executionContext) {
+            if (PUT_OBJECT.matches(method, false)) {
+                method = transformPutFileOverload(method);
```

Comment:
Don't we need to invoke super.visitMethodInvocation?

Reply:
We do, on 72. I don't think ordering matters here